central/image/datastore/store/v2/postgres/store.go

```diff
@@ -557,6 +560,17 @@ func (s *storeImpl) populateImage(ctx context.Context, tx *postgres.Tx, image *s
 			cveParts = append(cveParts, cvePart)
 		}
 
+		// TODO(ROX-27399):  Adding the index of where the vuln appeared in the component
+		// is not likely sustainable.  We cannot easily guarantee the order is the same when
+		// we pull the data out.  This sort is temporary to keep moving, and will be
+		// removed when the ID of the CVE is adjusted to no longer use the index of where
+		// the CVE occurs in the component list.
+		sort.SliceStable(cveParts, func(i, j int) bool {
```

Comment:
I was curious if the choice of adding the index of the CVE would be a problem.  I will revisit the key creation subsequently and then remove this.

Reply:
probably.  I think the original intent was to account for a component that had the same CVE more than once.  I'm not sure how likely that is or if we could simply use CVE + Component ID + hash of the CVE.  In the even the 2nd occurrence came from a different source or something.  We could use a simple sequence if we didn't convert back and forth between the scan data and the data we save for searching.  That is the crux of the issue that I didn't really catch until now.